### PR TITLE
Setting transition pending flag shouldn't be part of a surrounding transition

### LINKED ISSUE #26226: Bug: Nested useTransition makes isPending of outer one always false
Nested `startTransition` call "takes over" and makes parent `startTransition` unable to track `isPending`.

Seb says it's a bug.

Repro:

https://codesandbox.io/s/pensive-breeze-rg70wn?file=/IndexPage.js:192-288

1. Click the button
2. `isPending` in `IndexPage.js` is `true`

However, `isPending` in `App.js` is `false`.

Expected: `isPending` in `App.js` is also `true`.

### PR DESCRIPTION
**Pull request by sophiebits**
Fixes #26226. (Is this the right fix?)

### CODE REVIEW

**Review by gaearon — APPROVED** [review]
seems good to me but i’m surprised we had no tests that either test this or test the opposite

**Comment by sophiebits** [comment]
hmm does this change mean you can’t implement useTransition in user space?

File: packages/react-reconciler/src/__tests__/ReactTransition-test.js [hunk 1]
```diff
@@ -959,4 +959,50 @@ describe('ReactTransition', () => {
...
+          {', '}
+          <Text text={'B ' + isPendingB} />
+        </>
+      );
+    }
+    const root = ReactNoop.createRoot();
+    await act(async () => {
+      root.render(<App />);
+    });
+    expect(Scheduler).toHaveYielded([0, 'A false', 'B false']);
```
Comment (acdlite) [inline packages/react-reconciler/src/__tests__/ReactTransition-test.js:992]: Wanna update this to use `assertLog`? If not I can fix post merge.

**Review by acdlite — APPROVED** [review]
Good fix and test! I think this used to work but because we were missing the test it broke during a refactor

**Pull request merged**